templates/stacklist.hbs

```diff
@@ -14,9 +14,13 @@
                     </div>
                 {{/each}}
                 <div class="stacklist-button">
-                    <button class="stacklist-next" aria-label="{{_globals._accessibility._ariaLabels.next}}" role="button">
-                        {{{_button.startText}}}
-                    </button>
+                    {{#if _globals._components._stacklist.ariaButtonLabel}}
+                        <button class="stacklist-next" aria-label="{{_globals._components._stacklist.ariaButtonLabel}}" role="button">
+                    {{else}}
+                        <button class="stacklist-next" aria-label="{{_globals._accessibility._ariaLabels.next}}" role="button">
+                    {{/if}}
+                            {{{_button.startText}}}
+                        </button>
```

Comment:
This seems a bit bulky. I'd prefer something like
```
var ariaLabel = _globals._components._stacklist.ariaButtonLabel || _globals._accessibility._ariaLabels.next;
this.model.set({buttonAriaLabel: ariaLabel});
```
in the view, and then  `<button class="stacklist-next" aria-label="{{buttonAriaLabel}}" role="button">` in the template.

PS do we still need the `role="button"` attribute for a button element?!

Before changing this however, am I right in thinking that screen readers will prioritise reading the aria label over the button text? Or will both be read? If the former then I wouldn't want the actual text to be missed, especially since there is an option to change this for each item. Also, since you've now included a reference to the button in the instruction - would this be enough since there is only one button?

Reply:
You're right that the content of the button won't be read. What I'd really like to do is revise/append the `aria-label` with whatever the text of that specific button is going to be, since the start text can be different from the subsequent buttons, according to the component values. so that you're only using `_globals._accessibility._ariaLabels.next` as a fallback if the labels aren't dictated in components.json **and**_globals._components._stacklist.ariaButtonLabel is empty. I haven't yet looked into where that's being done in the js.